particula/aerosol_dynamics/parcel.py

```diff
@@ -48,43 +38,68 @@ def name(self) -> str:
 
     def create_and_add_particle(
-        self, name: str, radius, density=1e3, charge=0
+        self, name: str, radius, density=1e3 * u.kg / u.m**3,
+        charge=0 * u.dimensionless
     ):
         """creats and then Adds a particle to the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            name    (str)   [no units],
+            radius  (float) [m]
+        Optional:
+            density (float) [kg/m**3]       default = 1e3 [kg/m**3],
+            charge  (int)   [dimensionless] default = 0 [dimensionless]
         """
-        self._particle_data.append(
+
+        try:  # maybe move to particle class
+            radius.check('m')
+        except AttributeError:
+            print('Please add units to radius. E.g., radius = 150 * u.nm]')
+
+        self.add_particle(
             particle.Particle(name, radius, density, charge)
         )
 
     def create_and_add_list_of_particle(
-        self, name_of_particles, radius_of_particles
+        self, name_of_particles, radius_of_particles,
+        density_of_particles=None, charge_of_particles=None
     ):
-        """Adds a list of particles to the parcel based on size only.
+        """Adds a list of particles to the parcel based on size only
+            or delcaring denisty and charge.
         Parameters:
-            name_of_particles (list): The name of the particles.
-            radius_of_particles (list): The radius of the particle.
-            TODO: add charge and mass as passible parameters.
+            name_of_particles   (str)   [no units],
+            radius_of_particles (list)  [m]
+        Optional:
+            density_of_particles (list) [kg/m**3],
+            charge_of_particles  (list) [dimensionless]
         """
-        for size_add in radius_of_particles:
-            self.create_and_add_particle(name_of_particles, size_add)
-
-    def remove_particle(self, particle_data):
+        if density_of_particles is None:
+            for size_add in radius_of_particles:
+                self.create_and_add_particle(name_of_particles, size_add)
+        else:
+            for size_add, density_add, charge_add in zip(
+                    radius_of_particles, density_of_particles,
+                    charge_of_particles
+                    ):
+                self.create_and_add_particle(
+                    name_of_particles, size_add, density_add, charge_add
+                )
+
+    def remove_particle(self, particle_index):
         """Removes a particle from the parcel.
         Parameters:
-            particle_data (list): The radius of the particle.
+            particle_index (obj): int or array of ints
+            Indicate indices of sub-arrays to remove along the specified axis.
         """
-        self._particle_data.remove(particle_data)
+        self._particle_data = np.delete(self._particle_data, particle_index)
 
     def remove_all_particles(self):
```

Comment:
What would be the equivalent of `-> list` here?

Reply:
I don't know what you mean. It'll delete the indexes you give it using numpy?

Reply:
I was trying to figure out how to add this check for a numpy array. Your stuff is fine, don't worry about it. We will figure it out later.